In [1]:
import pandas as pd
import numpy as np
df = pd.read_csv(r'C:\Users\User\Downloads\FoodBalanceSheets_E_Africa_NOFLAG.csv',encoding="latin-1")

Answer the following questions based on the African food production dataset provided by the FAO website already provided

What is the total sum of Animal Fat produced in 2014 and 2017 respectively?

In [3]:
df1= df.groupby(['Item']).sum()
df1

Area Code  Item Code  Element Code      Y2014      Y2015  \
Item                                                                            
Alcohol, Non-Food        38447     752497       1515090    2403.00    2180.00   
Alcoholic Beverages      75016    1649136       2171381  102410.11   98783.72   
Animal Products          18060     397035         90990   11935.65   11811.73   
Animal fats              85279    1882494       2567303  209460.54  200675.72   
Apples and products      69280    1352989       1920418    9499.23   10559.15   
...                        ...        ...           ...        ...        ...   
Vegetables, Other        77779    1518715       2275409  155038.96  158104.08   
Vegetal Products         18060     391905         90990  107145.19  107064.17   
Wheat and products       87073    1639683       2654934  232670.13  234710.51   
Wine                     66206    1319535       1818328    4497.36    4251.81   
Yams                     50798     970905       1525745  200396.96  203151.78   

                         Y2016      Y2017      Y2018  
Item                                                  
Alcohol, Non-Food      2223.00    2348.00    2293.00  
Alcoholic Beverages   96958.75   95581.06   97847.27  
Animal Products       11661.69   11547.65   11578.61  
Animal fats          183314.22  269617.53  269648.27  
Apples and products    9853.26   10198.90    9640.51  
...                        ...        ...        ...  
Vegetables, Other    156218.90  157752.59  163987.21  
Vegetal Products     106792.72  107655.20  107775.39  
Wheat and products   239377.07  240047.62  242645.19  
Wine                   3872.09    4178.02    4039.32  
Yams                 220626.20  229174.59  221272.09  

[119 rows x 8 columns]

The total sum of Animal Fat produced in 2014 and 2017 is 209460.54 and 269617.53 respectively

What is the mean and standard deviation across the whole dataset for the year 2015 to 3 decimal places?

In [5]:
df['Y2015'].mean()


135.23596649549236

In [6]:
df['Y2015'].std()

1603.4039839423301

What is the total number and percentage of missing data in 2016 to 2 decimal places?

In [7]:
df['Y2016'].isnull().sum()

1535

In [8]:
df['Y2016'].isnull().sum()*100 / len(df['Y2016'])

2.518747025909456

Which year had the highest correlation with ‘Element Code’? 

In [10]:
 corr_2014 = df['Element Code'].corr(df['Y2014'])
corr_2015 = df['Element Code'].corr(df['Y2015'])
corr_2016 = df['Element Code'].corr(df['Y2016'])
corr_2017 = df['Element Code'].corr(df['Y2017'])
corr_2018 = df['Element Code'].corr(df['Y2018'])
print(corr_2014)
print(corr_2015)
print(corr_2016)
print(corr_2017)
print(corr_2018)

0.024456819495221795
0.02388919930163992
0.023443997584961428
0.02425400776004572
0.02427857195351955


Year 2014 has the highest correlation

What year has the highest sum of Import Quantity?

In [11]:
df2 = df.groupby(['Element']).sum()
df2

Area Code  Item Code  Element Code  \
Element                                                                      
Domestic supply quantity                   708993   14197445      28068795   
Export Quantity                            599910   11840553      26026133   
Fat supply quantity (g/capita/day)         675050   13535000       3435732   
Feed                                       176272    3538507       7282199   
Food                                       663295   13285035      25406622   
Food supply (kcal/capita/day)              674057   13511060       3329296   
Food supply quantity (kg/capita/yr)        658446   13185401       3163725   
Import Quantity                            688174   13795966      28834929   
Losses                                     274353    5424803      10292107   
Other uses (non-food)                      235554    4729749       8926728   
Processing                                 271940    5350416      10313310   
Production                                 526751   10450053      21388191   
Protein supply quantity (g/capita/day)     675050   13535000       3385502   
Residuals                                  623271   12421089      24066350   
Seed                                       103537    2035933       4211574   
Stock Variation                            571566   11329527      21464704   
Total Population - Both sexes                6020     112545         22995   
Tourist consumption                         50308    1486528       2869905   

                                             Y2014       Y2015       Y2016  \
Element                                                                      
Domestic supply quantity                1996716.35  2021493.55  2044842.70   
Export Quantity                          150020.64   157614.47   151920.46   
Fat supply quantity (g/capita/day)        10225.56    10235.74    10102.77   
Feed                                     216927.89   225050.22   228958.65   
Food                                    1212332.49  1232361.10  1247022.17   
Food supply (kcal/capita/day)            454257.00   453383.00   451810.00   
Food supply quantity (kg/capita/yr)       49650.63    49345.13    48985.28   
Import Quantity                          274144.48   267018.46   286582.78   
Losses                                   153223.00   155439.00   157787.00   
Other uses (non-food)                     78718.13    66254.41    69563.68   
Processing                               282923.00   287929.00   280631.00   
Production                              1931287.75  1947019.39  1943537.15   
Protein supply quantity (g/capita/day)    11836.46    11833.95    11779.69   
Residuals                                 30149.00    30045.00    37224.00   
Seed                                      21922.92    23976.82    23389.20   
Stock Variation                           58749.83    34910.99    33140.12   
Total Population - Both sexes           1031585.00  1058081.00  1085107.00   
Tourist consumption                         416.00      349.00       89.00   

                                             Y2017       Y2018  
Element                                                         
Domestic supply quantity                2088198.10  2161192.10  
Export Quantity                          182338.80   181594.80  
Fat supply quantity (g/capita/day)        10253.84    10258.69  
Feed                                     223705.68   233489.68  
Food                                    1258888.28  1303841.28  
Food supply (kcal/capita/day)            454681.00   455261.00  
Food supply quantity (kg/capita/yr)       48690.04    49056.85  
Import Quantity                          294559.09   287997.09  
Losses                                   160614.00   163902.00  
Other uses (non-food)                     91645.97    91300.97  
Processing                               292836.00   308429.00  
Production                              2030056.89  2075072.89  
Protein 

From the results above, Year 2017 has the highest sum of import quantity

What is the total number of the sum of Production in 2014?

From the results above, 1931287.75

Which of these elements had the highest sum in 2018?

From the results above, Domestic supply quantity 

Which of these elements had the 3rd lowest sum in 2018?

From the results above, Protein supply quantity (g/capita/day)

What is the total Import Quantity in Algeria in 2018?

In [19]:
df3 = df.groupby(['Area','Element']).sum()
df3['Y2018'].head(18)

Area     Element                               
Algeria  Domestic supply quantity                  90023.42
         Export Quantity                             811.54
         Fat supply quantity (g/capita/day)          309.09
         Feed                                      10668.00
         Food                                      63736.33
         Food supply (kcal/capita/day)             13288.00
         Food supply quantity (kg/capita/yr)        1509.28
         Import Quantity                           36238.29
         Losses                                     7708.00
         Other uses (non-food)                      3970.08
         Processing                                 2479.00
         Production                                60766.67
         Protein supply quantity (g/capita/day)      367.28
         Residuals                                   630.00
         Seed                                        825.00
         Stock Variation                            

From the results above, 36238.29

What is the total number of unique countries in the dataset?

In [20]:
df.nunique()

Area Code         49
Area              49
Item Code        122
Item             119
Element Code      18
Element           18
Unit               5
Y2014           4493
Y2015           4520
Y2016           4520
Y2017           4537
Y2018           4591
dtype: int64

From the results above, the total number of unique countries is 49